deepchem/models/torch_models/layers.py

```diff
@@ -14,36 +15,175 @@ class ScaleNorm(nn.Module):
   Finally, the result is returned as `input_tensor * norm value`.
 
   This layer can be used instead of LayerNorm when a scaled version of the norm is required.
-  Instead of performing the scaling operation (scale / norm) in a lambda-like layer, we are defining it within this layer to make prototyping more efficient.
+  Instead of performing the scaling operation (`scale / norm`) in a lambda-like layer, we are defining it within this layer to make prototyping more efficient.
 
   References
   ----------
   .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
-  
+
   Examples
   --------
   >>> import deepchem as dc
   >>> scale = 0.35
-  >>> layer = dc.models.torch_models.ScaleNorm(scale)
+  >>> layer = dc.models.torch_models.layers.ScaleNorm(scale)
   >>> input_tensor = torch.tensor([[1.269, 39.36], [0.00918, -9.12]])
-  >>> output_tensor = layer.forward(input_tensor)
+  >>> output_tensor = layer(input_tensor)
   """
 
-  def __init__(self, scale, eps=1e-5):
+  def __init__(self, scale: int, eps: float = 1e-5):
     """Initialize a ScaleNorm layer.
 
     Parameters
     ----------
     scale: Real number or single element tensor
       Scale magnitude.
     eps: float
-      Epsilon value.
+      Epsilon value. Default = 1e-5.
     """
-
     super(ScaleNorm, self).__init__()
     self.scale = nn.Parameter(torch.tensor(math.sqrt(scale)))
     self.eps = eps
 
-  def forward(self, x):
+  def forward(self, x: torch.Tensor):
     norm = self.scale / torch.norm(x, dim=-1, keepdim=True).clamp(min=self.eps)
     return x * norm
+
+
+class MATEmbedding(nn.Module):
+  """Embedding layer to create embedding for inputs.
+
+  In an embedding layer, input is taken and converted to a vector representation for each input.
+  In the MATEmbedding layer, an input tensor is processed through a dropout-adjusted linear layer and the resultant vector is returned.
+
+  References
+  ----------
+  .. [1] Lukasz Maziarka et al. "Molecule Attention Transformer" Graph Representation Learning workshop and Machine Learning and the Physical Sciences workshop at NeurIPS 2019. 2020. https://arxiv.org/abs/2002.08264
+
+  Examples
+  --------
+  >>> import deepchem as dc
+  >>> layer = MATEmbedding(d_input = 1024, d_output = 1024, dropout_p = 0.2)
```

Comment:
Can you show a usage example of how to invoke the layer on sample input?

Reply:
Added.